lib/models/asset-size-printer.js

```diff
@@ -63,16 +63,16 @@ module.exports = class AssetPrinterSize {
         })
         // Print human-readable file sizes (including gzipped)
         .map((file) => {
-          let contentsBuffer = fs.readFileSync(file);
-          return gzip(contentsBuffer).then((buffer) => ({
-            name: file,
-            size: contentsBuffer.length,
-            gzipSize: buffer.length,
-            showGzipped: contentsBuffer.length > 0,
-          }));
+          return pool
+            .exec('gzipStats', [file])
+            .then((result) => result)
+            .catch((err) => console.error(err));
         });
 
-      return resolve(Promise.all(assets));
+      return Promise.all(assets).then((data) => {
+        pool.terminate(); // terminate all workers when done
```

Comment:
This should probably be in a `.finally`

Reply:
done

Reply:
It seems like the test is failing if I remove the new promise wrapper so I reverted only that change.